makefile

```diff
@@ -27,16 +27,64 @@ MISSPELL_CORRECTION=misspell -w
 STATICCHECK=staticcheck
 IMPI=impi
 
-
 .PHONY: all
 all: check
-	$(MAKE) generate
+	$(MAKE) generate-sapm
 	$(MAKE) test
 
-.PHONY: generate
-generate:
-	mkdir -p gen
-	docker run --rm -v $(PWD):$(PWD) -w $(PWD) znly/protoc --gofast_out=./gen/ -I./ -I./vendor/github.com/gogo/protobuf/ -I./vendor/ sapm.proto
+# Target directory to write generated files to.
+PROTO_TARGET_GEN_DIR=./gen
+
+.PHONY: generate-sapm
+generate-sapm:
+	mkdir -p $(PROTO_TARGET_GEN_DIR)
+	docker run --rm -v $(PWD):$(PWD) -w $(PWD) znly/protoc --gofast_out=$(PROTO_TARGET_GEN_DIR)/ -I./ -I./vendor/github.com/gogo/protobuf/ -I./vendor/ sapm.proto
```

Comment:
didn't you move sapm.proto to proto directory?

Reply:
Independent PR. I will rebase when the other PR is merged